### DISCUSSION TITLE
SDK 1.9 and result forms / mismatch between rules and control rules

### DISCUSSION THREAD

**Discussion by jcagnolo** [OP-TED/eForms-SDK#994] [Q&A] [answered]
Hello, 

We have developed our application in SDK 1.9 and are experiencing difficulties with the results forms (F25, F29 and higher). 
Do you manage to send result forms without attribution (field value "BT-142-LotResult" with "No winner has been chosen and the competition is closed")? 
And are you successful in sending forms with an award but no subcontractor (field value "BT-142-LotResult" with "At least one winner has been chosen" and fields ID - Subcontractor (OPT-301-Tenderer-SubCont) empty and ID - Main Contractor (OPT-301-Tenderer-MainCont) empty)? 
In the latter case, we know that there is a gap in the SDK between the error messages provided for in SDK 1.9 and the business control rules. Do you know if this issue will be corrected? 

Best regards, 
Jeanne 

Translated with DeepL.com (free version)

**Comment by YvesJo** [comment]
Hi,
The provided information is lacking some precision in order to be able to answer the questions.
Please, note that for forms 25 to 28, BT-142-LotResult is forbidden and you will never be able to submit successfully these forms if they contain that field.
If you need some help detecting the issues, please get in touch with the helpdesk and provide the necessary information.
KR

**Comment by jcagnolo** [comment]
Hello Yves, 

Indeed, this problem concerns forms F29 and higher. 

We have carried out the following tests based on an F29 form, with identical data except for the result section, bidder sub-section and the BT-142-LotResult field. We obtain the following results: 
- When we enter the value "No winner has been chosen and the competition is closed" (BT-142-LotResult), the notice is sent and published; 
- When we enter the value "At least one winner has been chosen" (BT-142-LotResult), we are obliged to declare a main contractor (OPT-301-Tenderer-MainCont). 
o In this case, after entering a main contractor in the Result sub-section : 
 If we do not enter a subcontractor (OPT-301-Tenderer-SubCont), our notice is rejected; 
 If we enter a subcontractor (OPT-301-Tenderer-SubCont), our notice is validated. 

This situation poses a problem for us because, under French law, we need to be able to publish award notices without a subcontractor.  In this context, can you tell us whether this situation is normal for you? And whether compliance with French law is possible, or whether the difficulty lies in the way our developers have implemented the rules? 

I've also contacted TED support with the same message. I'm waiting for them to get back to me. 

Thank you,
Jeanne

***

Bonjour Yves, 

Effectivement, cette difficulté concerne les formulaires F29 et plus. 

Nous avons effectué les tests suivants sur la base d’un formulaire F29, avec des données identiques à l’exception de la section résultat, sous section soumissionnaire et du champ BT-142-LotRésult. Nous obtenons les résultats suivants : 
-	Lorsque nous saisissons la valeur « Aucun lauréat n’a été choisi et le concours est clos » (BT-142-LotRésult), l’avis est envoyé et publié ; 
-	Lorsque nous saisissons la valeur « Au moins un lauréat a été choisi » (BT-142-LotResult) nous sommes obligés de déclarer un contractant principal (OPT-301-Tenderer-MainCont). 
o	Dans ce cas et après avoir saisi un contractant principal dans la sous-section Résultat : 
	Si nous ne saisissons pas de sous-traitant (OPT-301-Tenderer-SubCont), notre avis est refusé ; 
	Si nous saisissons un sous-traitant (OPT-301-Tenderer-SubCont), notre avis est validé. 

Cette situation nous pose un problème parce qu’au regard du droit français, il nous est nécessaire de pouvoir publié des avis d’attribution sans sous-traitant.  Dans ce contexte, pouvez-vous nous indiquer s’il est normal pour vous que nous ayons cette situation ? Et si une mise en conformité au regard du droit français est possible ou bien si la difficulté vient de la manière dont nos développeurs ont implémenté les règles ? 

De plus, j’ai également contacté l’assistance depuis TED avec le même message. Je suis dans l’attente de leur retour. 

Merci,
Jeanne

**Reply by bertrand-lorentz** [reply]
Hello,
It seems that the following example from version 1.9.3 of the SDK corresponds to the situation you describe:
https://github.com/OP-TED/eForms-SDK/blob/1.9.3/examples/notices/can_24_minimal.xml#L66

BT-142-LotResult set to "At least one winner was chosen" (selec-w), and no subcontractor (OPT-301-Tenderer-SubCont) indicated.

This example is valid for the rules in place for SDK 1.9.3. So there does not seem to be any rule that requires the indication of a subcontractor.

There must be something else that causes the behavior you describe.
Could you attach an XML notice that is rejected, and indicate what is the error you get when it is rejected ?

**Comment by jcagnolo** [comment]
Hello,

Thank you for your quick feedback. You will find in PJ the XML of blocked notices (if they are not attached, can you send me your email as github doesn't seem to accept this format?).
I'm going to offer the XML sent for reading to our developers, but I'd be delighted to receive feedback on what I've sent you.

Best regards,
Jeanne

Bonjour,

Merci pour votre retour rapide. Vous trouverez en PJ les XML d’avis bloqués (s’ils ne sont pas joint, pouvez-vous me transmettre votre email car github ne semble pas accepter ce format ?).
Je vais proposer le XML envoyé en lecture à nos développeurs, mais je suis preneuse d’un retour sur ce que je vous ai transmis.

Bien à vous,
Jeanne


***@***.***
Jeanne CAGNOLO
Analyste – Secteur public
Mobile : +33 (0)6 99 09 85 99
***@***.******@***.***>
www.wavestone.com<https://www.wavestone.com/>

Tour Franklin : 100 - 101 terrasse Boieldieu
92042 Paris La Défense Cedex
--


De : Bertrand Lorentz ***@***.***>
Envoyé : mardi 13 août 2024 14:27
À : OP-TED/eForms-SDK ***@***.***>
Cc : CAGNOLO Jeanne ***@***.***>; Author ***@***.***>
Objet : Re: [OP-TED/eForms-SDK] SDK 1.9 and result forms / mismatch between rules and control rules (Discussion #994)


Hello,
It seems that the following example from version 1.9.3 of the SDK corresponds to the situation you describe:
https://github.com/OP-TED/eForms-SDK/blob/1.9.3/examples/notices/can_24_minimal.xml#L66<https://urldefense.com/v3/__https:/github.com/OP-TED/eForms-SDK/blob/1.9.3/examples/notices/can_24_minimal.xml*L66__;Iw!!NEMsmePo_HYI!Zumz8fTyAziTlWGndQTuzRLhSVKvDv17_2w3irNV_6wHz0JwVHXNIWAcIApOx96D5eBOhYwsPwKbfAWX9RuTzwqas51gnGw$>

BT-142-LotResult set to "At least one winner was chosen" (selec-w), and no subcontractor (OPT-301-Tenderer-SubCont) indicated.

This example is valid for the rules in place for SDK 1.9.3. So there does not seem to be any rule that requires the indication of a subcontractor.

There must be something else that causes the behavior you describe.
Could you attach an XML notice that is rejected, and indicate what is the error you get when it is rejected ?

—
Reply to this email directly, view it on GitHub<https://urldefense.com/v3/__https:/github.com/OP-TED/eForms-SDK/discussions/994*discussioncomment-10325400__;Iw!!NEMsmePo_HYI!Zumz8fTyAziTlWGndQTuzRLhSVKvDv17_2w3irNV_6wHz0JwVHXNIWAcIApOx96D5eBOhYwsPwKbfAWX9RuTzwqaKtG_VqU$>, or unsubscribe<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/BFZWK7VJFSGYDK5L3HQIFSTZRH3S3AVCNFSM6AAAAABMIDTAYWVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAMZSGU2DAMA__;!!NEMsmePo_HYI!Zumz8fTyAziTlWGndQTuzRLhSVKvDv17_2w3irNV_6wHz0JwVHXNIWAcIApOx96D5eBOhYwsPwKbfAWX9RuTzwqaavq2GgY$>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

The information transmitted in the present email including the attachment is intended only for the person to whom or entity to which it is addressed and may contain confidential and/or privileged material. Any review, retransmission, dissemination or other use of, or taking of any action in reliance upon this information by persons or entities other than the intended recipient is prohibited. If you received this in error, please contact the sender and delete all copies of the material.

Ce message et toutes les pièces qui y sont éventuellement jointes sont confidentiels et transmis à l'intention exclusive de son destinataire. Toute modification, édition, utilisation ou diffusion par toute personne ou entité autre que le destinataire est interdite. Si vous avez reçu ce message par erreur, nous vous remercions de nous en informer immédiatement et de le supprimer ainsi que les pièces qui y sont éventuellement jointes.

~w1Y2c3è4`

**Reply by YvesJo** [reply, marked as answer]
Hi,
Thanks for the details.
You may also have a look at the following discussion: https://github.com/OP-TED/eForms-SDK/discussions/917 .
Please note that OPT-301-Tenderer-MainCont does not identify "the contractor" but "the contractor, the subcontractor is providing services to". The contractors are identified with the field OPT-300-Tenderer instead.
